GiniVision/Classes/Core/CapturedImagesStackView.swift

```diff
@@ -0,0 +1,178 @@
+            return CGSize(width: 40, height: 60)
+        } else {
+            return CGSize(width: 30, height: 45)
+        }
+    }()
+    var didTapImageStackButton: (() -> Void)?
+    fileprivate let stackCountCircleSize = CGSize(width: 25, height: 25)
+    fileprivate var imagesCount: Int = 0
+    
+    fileprivate lazy var thumbnailButton: UIButton = {
+        let button = UIButton()
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.layer.shadowColor = UIColor.black.cgColor
+        button.layer.shadowRadius = 1
+        button.layer.shadowOpacity = 0.5
+        button.layer.shadowOffset = CGSize(width: -2, height: 2)
+        button.addTarget(self, action: #selector(thumbnailButtonAction), for: .touchUpInside)
+        
+        return button
+    }()
+    
+    lazy var thumbnailStackBackgroundView: UIView = {
+        let view = UIView()
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.backgroundColor = .lightGray
+        view.isHidden = true
+        return view
+    }()
+    
+    fileprivate lazy var stackIndicatorLabel: UILabel = {
+        var label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.textColor = Colors.Gini.blue
+        return label
+    }()
+    
+    fileprivate lazy var stackIndicatorCircleView: UIView = {
+        var view = UIView(frame: .zero)
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.frame.size = self.stackCountCircleSize
+        view.backgroundColor = .white
+        view.layer.cornerRadius = self.stackCountCircleSize.width / 2
+        view.layer.shadowRadius = 1
+        view.layer.shadowOpacity = 0.3
+        view.layer.shadowOffset = CGSize(width: -1, height: 1)
+        return view
+    }()
+    
+    fileprivate lazy var capturedImagesStackSubtitleLabel: UILabel = {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.text = NSLocalizedString("ginivision.camera.capturedImagesStackLabel",
+                                       bundle: Bundle(for: GiniVision.self),
+                                       comment: "label shown below images stack")
+        label.textAlignment = .center
+        label.textColor = .white
+        label.font = label.font.withSize(12)
+
+        return label
+    }()
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        addSubview(thumbnailStackBackgroundView)
+        addSubview(thumbnailButton)
+        addSubview(stackIndicatorCircleView)
+        addSubview(capturedImagesStackSubtitleLabel)
+        stackIndicatorCircleView.addSubview(stackIndicatorLabel)
+        addConstraints()
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("Use init(frame:) initializer instead")
+    }
+
+}
+
+// MARK: - Public methods
+
+extension CapturedImagesStackView {
+    func absoluteThumbnailFrame(from view: UIView) -> CGRect {
```

Comment:
Initially, I didn't like this method. It is confusing and breaks encapsulation. I understand that there's no better way to create the transition animation, but I think it at least warrants a comment.

Reply:
Don't understand this. Why does it break encapsulation? `CapturedImagesStackView` is providing a way of getting its position from other view's perspective.

Reply:
It's not really "absolute". It's not relative to the screen, but to a superview.

I just thought that it was a strange method. I'd put a comment saying that it's related to animations and you can get the frame of the thumbnail so you can use it for animating. But just a remark from my side, not an issue.

Reply:
I'm sorry, but I'm still not satisfied. The comment is too generic - I see what the method does, but I don't understand why it exists. I'd say, even though is not super loosely coupled, just write that it's about being able to make a transition from the thumbnail to a new screen.

Also, as I pointed out, you are not returning an absolute frame, but rather a frame relative to another view. So the name is misleading. I'd use `frameRelative:to:` or something.